cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -222,4 +227,51 @@ private void updateTetherStatus(HttpResponder responder, String peer, TetheringS
     }
     responder.sendStatus(HttpResponseStatus.OK);
   }
+
+  /**
+   * Processes the list of tethering program updates received from the client and updates the program statuses
+   */
+  private void processProgramUpdates(String content) {
+    if (content.isEmpty()) {
+      return;
+    }
+    TetheringProgramUpdate programUpdate = GSON.fromJson(content, TetheringProgramUpdate.class);
+    for (TetheringProgramUpdate.TetheringProgramUpdateRequest request : programUpdate.getUpdateRequests()) {
+      ProgramRunId receivedProgram = GSON.fromJson(request.getProgramRunId(), ProgramRunId.class);
+      ProgramRunId programRunId = new ProgramRunId(NamespaceId.SYSTEM.getNamespace(), receivedProgram.getApplication(),
```

Comment:
not sure what namespace to use here since the received programRunId will be referencing a namespace from the tethered instance. Is there a way to find the namespace by the runId?